cmake/FindR.cmake

```diff
@@ -46,42 +46,67 @@ if(R_COMMAND)
   set(OLD_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES})
   set(CMAKE_FIND_LIBRARY_SUFFIXES ${CMAKE_FIND_LIBRARY_SUFFIXES} ".dll")
 
-  execute_process(WORKING_DIRECTORY .
-                  COMMAND ${R_COMMAND} RHOME
-                  OUTPUT_VARIABLE R_ROOT_DIR
-                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-
-  set(R_HOME ${R_ROOT_DIR} CACHE PATH "R home directory obtained from R RHOME")
-  
-  execute_process(WORKING_DIRECTORY .
-                  COMMAND ${R_COMMAND} CMD config --ldflags
-                  OUTPUT_VARIABLE R_LDFLAGS
-                  OUTPUT_STRIP_TRAILING_WHITESPACE)
-  set(R_LDFLAGS ${R_LDFLAGS} CACHE PATH "R CMD config --ldflags")
-
-  find_path(R_INCLUDE_DIR R.h
-            HINTS ${R_ROOT_DIR} ${R_ROOT_DIR}/bin/${R_LIB_ARCH}
-            PATHS /usr/local/lib /usr/local/lib64 /usr/share
-            PATH_SUFFIXES include R/include
-            DOC "Path to file R.h")
-
-  find_library(R_LIBRARY_BASE R
-            HINTS ${R_ROOT_DIR}/lib ${R_ROOT_DIR}/bin/${R_LIB_ARCH}
-            DOC "R library (example libR.a, libR.dylib, etc.).")
-
-  find_library(R_LIBRARY_BLAS NAMES Rblas blas
-            HINTS ${R_ROOT_DIR}/lib ${R_ROOT_DIR}/bin/${R_LIB_ARCH}
-            DOC "Rblas library (example libRblas.a, libRblas.dylib, etc.).")
-
-  find_library(R_LIBRARY_LAPACK NAMES Rlapack lapack
-            HINTS ${R_ROOT_DIR}/lib ${R_ROOT_DIR}/bin/${R_LIB_ARCH}
-            DOC "Rlapack library (example libRlapack.a, libRlapack.dylib, etc.).")
-
-  find_library(R_LIBRARY_READLINE readline
-            DOC "(Optional) system readline library. Only required if the R libraries were built with readline support.")
+  if(CMAKE_SYSTEM_NAME STREQUAL "Emscripten")
+    message(STATUS "Configuring for Emscripten...")
+
+    set(R_HOME "${CMAKE_PREFIX_PATH}/lib/R" CACHE PATH "R home directory for Emscripten")
+
+    set(R_INCLUDE_DIR "${R_HOME}/include" CACHE PATH "Path to R include directory")
+    file(GLOB_RECURSE R_INCLUDE_FILES "${R_INCLUDE_DIR}/*")
+    message(STATUS "Contents of R include directory:")
+    foreach(file IN LISTS R_INCLUDE_FILES)
+        message(STATUS " - ${file}")
+    endforeach()
+
+    set(R_LDFLAGS "-Wl,--export-dynamic -fopenmp -Wl,-Bsymbolic-functions -flto=auto -ffat-lto-objects -flto=auto -Wl,-z,relro -L${R_HOME}/lib -lR -lpcre2-8 -ldeflate -llzma -lbz2 -lz -ltirpc -lrt -ldl -lm -licuuc -licui18n" CACHE STRING "Linker flags for R libraries in Emscripten")
+
+    set(R_LIBRARY_BASE "${R_HOME}/lib/libR.so" CACHE FILEPATH "R library (libR.so) in Emscripten")
```

Comment:
Well I think, this faulty. The r-base recipe provides use with a [libR.a](https://github.com/emscripten-forge/recipes/actions/runs/11127430850/job/30919715167#step:9:3083) instead of a libR.so (I might have been confused becasue we get back `libRblas.so` and `libRlapack.so` (but a `libR.a`)

Reply:
And I guess we use `target_link_libraries(${target_name} PUBLIC ${XEUS_R_XEUS_TARGET} PRIVATE ${R_LIBRARY_BASE})` instead of what we have below ?

https://github.com/jupyter-xeus/xeus-r/blob/cfdaf75d94563898aebae6f2640ff8ab789c9013/CMakeLists.txt#L213

Reply:
Good catch. Yes, it should be *libR.a*, *libRblas.so*, and *libRlapack.so*.